sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/xds/utils/RdsProtocolTransformer.java

```diff
@@ -215,4 +259,135 @@ private static XdsClusterWeight parseClusterWeight(ClusterWeight clusterWeight)
+                    parseTokenBucket(struct.getFieldsOrThrow("token_bucket").getStructValue());
+            optionalTokenBucket.ifPresent(xdsRateLimit::setTokenBucket);
+        }
+        if (struct.containsFields("filter_enforced")) {
+            Optional<io.sermant.core.service.xds.entity.FractionalPercent> optionalXdsFractionalPercent =
+                    parseRuntimeFractionalPercent(struct.getFieldsOrThrow("filter_enforced").getStructValue());
+            optionalXdsFractionalPercent.ifPresent(xdsRateLimit::setPercent);
+        }
+        if (struct.containsFields("filter_enabled")
+                && (xdsRateLimit.getPercent() == null || xdsRateLimit.getPercent().getNumerator() == 0)) {
+            Optional<io.sermant.core.service.xds.entity.FractionalPercent> optionalXdsFractionalPercent =
+                    parseRuntimeFractionalPercent(struct.getFieldsOrThrow("filter_enabled").getStructValue());
+            optionalXdsFractionalPercent.ifPresent(xdsRateLimit::setPercent);
+        }
+        List<XdsHeaderOption> responseHeaders = parseHeaderValueOptions(struct);
+        xdsRateLimit.setResponseHeaderOption(responseHeaders);
+        return xdsRateLimit;
+    }
+
+    private static List<XdsHeaderOption> parseHeaderValueOptions(Struct struct) {
+        if (!struct.containsFields("response_headers_to_add")) {
+            return Collections.EMPTY_LIST;
+        }
+        List<XdsHeaderOption> responseHeaders = new ArrayList<>();
+        ListValue headers = struct.getFieldsOrThrow("response_headers_to_add").getListValue();
+        for (Value value : headers.getValuesList()) {
+            Map<String, Value> headersMap = value.getStructValue().getFieldsMap();
+            if (headersMap.get("header") == null) {
+                continue;
+            }
+            Struct headerStruct = headersMap.get("header").getStructValue();
+            XdsHeaderOption responseHeader = new XdsHeaderOption();
+            if (headersMap.get("append") != null) {
+                responseHeader.setEnabledAppend(headerStruct.getFieldsOrThrow("append").getBoolValue());
+            }
+            if (!headerStruct.containsFields("key")) {
+                continue;
+            }
+            XdsHeader xdsHeader = new XdsHeader();
+            xdsHeader.setKey(headerStruct.getFieldsOrThrow("key").getStringValue());
+            if (headerStruct.containsFields("value")) {
+                xdsHeader.setValue(headerStruct.getFieldsOrThrow("value").getStringValue());
+            }
+            responseHeader.setHeader(xdsHeader);
+            responseHeaders.add(responseHeader);
+        }
+        return responseHeaders;
+    }
+
+    private static Optional<io.sermant.core.service.xds.entity.FractionalPercent> parseRuntimeFractionalPercent(
+            Struct filterEnabledStruct) {
+        Map<String, Value> valueMap = filterEnabledStruct.getFieldsMap();
+        if (valueMap.get("default_value") == null) {
```

Comment:
valueMap.get("default_value") can be assigned to a variable

Reply:
done